pkg/distributor/distributor.go

```diff
@@ -1029,3 +1036,27 @@ func (d *Distributor) AllUserStats(ctx context.Context) ([]UserIDStats, error) {
 
 	return response, nil
 }
+
+func (d *Distributor) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if d.distributorsRing != nil {
+		d.distributorsRing.ServeHTTP(w, req)
+	} else {
+		var unshardedPage = `
+			<!DOCTYPE html>
+			<html>
+				<head>
+					<meta charset="UTF-8">
+					<title>Cortex Distributor Status</title>
+				</head>
+				<body>
+					<h1>Cortex Distributor Status</h1>
+					<p>Distributor is not running with global limits disabled</p>
+				</body>
+			</html>`
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(unshardedPage))
+		if err != nil {
+			level.Error(d.log).Log("msg", "unable to serve status page", "err", err)
+		}
+	}
```

Comment:
Is this worth moving to `WriteHTMLResponse` in `pkg/util/http.go`?

Also there is no need to log the error while writing response. It's likely that client has disconnected, at that point, we cannot do anything about it.

Reply:
Yeah it makes sense so I've created that method and now we're using it in rules and distributor :)